recipes/dice-template-library/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

It looks like [your cmake creates a `Dice::dice-template-library` target](https://github.com/dice-group/dice-template-library/blob/ac08de8ce15456c55f60c5e2500abd5db4ec6c50/CMakeLists.txt#L18-L19).
But does not install one.
So you need to resort to the standard conan generated one (=`dice-template-library::dice-template-library`).

Using global variables for passing includes + libraries is being phased out, so you should do:
```suggestion
conan_basic_setup(TARGETS)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE dice-template-library::dice-template-library)
```

---

I think I figured it out. It seems to require both generators: "cmake" for the basic setup and "cmake_find_package" for the find_package part.

---

fixed